juriscraper/pacer/iquery.py

```diff
@@ -0,0 +1,187 @@
+        #         Corporation et al
+        #     <BR>
+        #       Mark L. Wolf, presiding
+        #     <BR>
+        #       <B>Date filed:</B> 02/10/2011
+        #     <BR>
+        #       <B>Date terminated:</B> 06/23/2014
+        #     <BR>
+        #       <B>Date of last filing:</B> 06/06/2018
+        #     <BR>
+        #   </CENTER>
+        #
+        # There's a bit more structured data in bankruptcy cases;
+        # but note the extra <B></B> pair in the first line:
+        #
+        # <div id="cmecfMainContent">
+        #   <input type="hidden" id="cmecfMainContentScroll" value="0">
+        #   <CENTER>
+        #       <B><FONT SIZE=+1>18-11572</FONT></B>
+        #       <B></B>Nancy Jean Stevens
+        #     <BR>
+        #       <B>Case type:</B> bk
+        #       <B>Chapter:</B> 7
+        #       <B>Asset:</B> No
+        #       <B>Vol: </B> v
+        #       <b>Judge:</b> Frank J. Bailey
+        #     <BR>
+        #       <B>Date filed:</B> 04/30/2018
+        #       <B>Date of last filing:</B> 06/04/2018
+        #     <BR>
+        #   </CENTER>
+        #
+        # There is some BK variation, both in terms of preence of
+        # docket numbers suffixes, and how the Judge: field name
+        # varies depending on the judge's chief judge status:
+        #
+        # <div id="cmecfMainContent">
+        #   <input type="hidden" id="cmecfMainContentScroll" value="0">
+        #   <CENTER>
+        #       <B><FONT SIZE=+1>18-11573</FONT></B>
+        #       <B></B>Sara A Taylor
+        #     <BR>
+        #       <B>Case type:</B> bk
+        #       <B>Chapter:</B> 7
+        #       <B>Asset:</B> No
+        #       <B>Vol: </B> v
+        #       <b>Chief Judge:</b> Melvin S. Hoffman
+        #     <BR>
+        #       <B>Date filed:</B> 04/30/2018
+        #       <B>Date of last filing:</B> 06/06/2018
+        #     <BR>
+        #   </CENTER>
+
+        # Rather than take the approach used by DocketParser of
+        # searching throughout the document for a docket number by
+        # regular expression, instead we take the <center> tag that follows
+        # the relevant cmecfMainContent <div>, and go line-by-line,
+        # delimited by <br>s. This approach is more structured:
+
+        center = self.tree.xpath('.//div[@id="cmecfMainContent"]//center')[0]
+        rows = self.redelimit_p(center, self.BR_REGEX)
+
+        # The first row demands special handling:
+        #   <B><FONT SIZE=+1>18-11572</FONT></B><B></B>Nancy Jean Stevens
+        # We take the docket number from the <font> tag (the innermost tag),
+        # although we could but have chosen the first <b> tag.
+        docket_number = force_unicode(rows[0].find('.//font').text_content())
+        # And case caption following the final <b></b> pair.
+        case_name_raw = force_unicode(rows[0].find('.//b[last()]').tail)
+
+        # Our job as a parser is to return the data, not to filter, clean,
```

Comment:
...editorializing here... :) Maybe remove this and just supply both `case_name_raw` and `case_name`?

Reply:
Again, because we **are** explaining.